dialects/matrix/passes/lower_matrix_mediumlevel.cpp

```diff
@@ -250,7 +250,7 @@ Ref LowerMatrixMediumLevel::rewrite_(Ref def) {
             iterator[idx]               = world.call<core::bitcast>(world.type_idx(dim_nat_def), iter);
             auto [new_mem, new_mat]     = new_acc->projs<2>();
             acc                         = {new_mem, new_mat};
-            current_mut->set(dim_nat_def, for_call);
+            current_mut->set(false, for_call); // TODO correct filter?
```

Comment:
The `dim_nat_ def?` filter was wrong. 
`false` is okay. But `true` as a filter should also work.
These functions just call the for-loop (they are just wrappers).
Therefore, we could always unfold the wrapper.